src/components/ListItem/ListItem.tsx

```diff
@@ -72,57 +76,72 @@ const ListItem: FC<ListItemProps> & { sizes?: typeof SIZES } = forwardRef(
       size = SIZES.SMALL,
       tabIndex = 0,
       children,
+      listId,
+      index,
+      updateSelectedItem,
       "data-testid": dataTestId
     },
     ref
   ) => {
+    const overrideId = id || `${listId}-item-${index}`;
     const componentRef = useRef(null);
     const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+    const prevSelected = usePrevious(selected);
+
+    useEffect(() => {
+      if (selected) {
+        updateSelectedItem?.(overrideId);
+      }
+      if (prevSelected && !selected) {
+        updateSelectedItem?.(null);
+      }
+    }, [overrideId, updateSelectedItem, selected, prevSelected]);
 
     const componentOnClick = useCallback(
       (event: React.MouseEvent) => {
         if (disabled) return;
-        onClick(event, id);
+        onClick(event, overrideId);
       },
-      [disabled, onClick, id]
+      [disabled, onClick, overrideId]
     );
 
     const onKeyDown = useCallback(
       (event: React.KeyboardEvent) => {
         if (disabled) return;
         const KEYS = [keyCodes.ENTER, keyCodes.SPACE];
         if (KEYS.includes(event.key)) {
-          onClick(event, id);
+          onClick(event, overrideId);
         }
       },
-      [disabled, onClick, id]
+      [disabled, onClick, overrideId]
     );
 
     const componentOnHover = useCallback(
       (event: React.MouseEvent | React.FocusEvent) => {
         if (disabled) return;
-        onHover(event, id);
+        onHover(event, overrideId);
       },
-      [disabled, onHover, id]
+      [disabled, onHover, overrideId]
     );
 
     return (
       <Text
-        element="div"
+        element="li"
         data-testid={dataTestId || getTestId(ComponentDefaultTestId.LIST_ITEM, id)}
         ref={mergedRef}
         className={cx(styles.listItem, className, getStyle(styles, camelCase(size)), {
           [styles.selected]: selected && !disabled,
           [styles.disabled]: disabled
         })}
-        id={id}
+        id={overrideId}
+        key={overrideId}
```

Comment:
is `key` needed?

Reply:
Yeah, list elements should have a key attribute - was receiving a warning about it